modules/exploits/linux/http/zyxel_parse_config_rce.rb

```diff
@@ -0,0 +1,128 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Zyxel parse_config.py Command Injection',
+        'Description' => %q{
+          This module exploits vulnerabilities in multiple Zyxel Products including VPN and USG devices (hopefully,
+          this is still in draft at the time of writing). The affected firmware version is 5.21 thru to 5.36.
+        },
+        'Author' => [
+          'SSD Secure Disclosure technical team', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'URL', 'https://ssd-disclosure.com/ssd-advisory-zyxel-vpn-series-pre-auth-remote-command-execution/'],
+          [ 'CVE', '2023-33012']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['linux', 'unix'],
+        'Privileged' => true,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-01-24',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
```

Comment:
I don't think this is true based on your exploit and my experience developing an exploit for this target against real hardware. I describe the issue [here](https://vulncheck.com/blog/zyxel-cve-2023-33012#you-get-one-shot).

Reply:
After chatting with Spencer I made some slight changes to what I suggested above and pushed it up. I think it's about ready to be tested.